server/src/main/java/io/crate/planner/node/ddl/CreateRolePlan.java

```diff
@@ -53,19 +69,77 @@ public StatementType type() {
         }
 
-        roleManager.createRole(createRole.roleName(), createRole.isUser(), newPassword)
+        // Map is always not null to simplify streaming.
+        // It's safe to create a JwtProperties object as user input already passed validation.
+        Map<String, Object> jwtPropertiesMap = Maps.getOrDefault(properties, JWT_PROPERTY_KEY, Map.of());
+        JwtProperties jwtProperties = null;
+        if (jwtPropertiesMap.isEmpty() == false) {
+            String iss = Maps.get(jwtPropertiesMap, "iss");
+            String username = Maps.get(jwtPropertiesMap, "username");
+            jwtProperties = new JwtProperties(iss, username);
+        }
+
+        roleManager.createRole(createRole.roleName(), createRole.isUser(), newPassword, jwtProperties)
             .whenComplete(new OneRowActionListener<>(consumer, rCount -> new Row1(rCount == null ? -1 : rCount)));
     }
+
+    public static Map<String, Object> validate(GenericProperties<Symbol> genericProperties,
+                                               TransactionContext txnCtx,
+                                               NodeContext nodeContext,
+                                               Row params,
+                                               SubQueryResults subQueryResults) {
+        Function<? super Symbol, Object> eval = x -> SymbolEvaluator.evaluate(
+            txnCtx,
+            nodeContext,
+            x,
+            params,
+            subQueryResults
+        );
+        Map<String, Object> properties = genericProperties.map(eval).properties();
+        for (var entry : properties.entrySet()) {
+            var key = entry.getKey();
+            if (PASSWORD_PROPERTY_KEY.equals(key) == false && JWT_PROPERTY_KEY.equals(key) == false) {
+                throw new IllegalArgumentException(
+                    String.format(Locale.ENGLISH, "\"%s\" is not a valid user property", key)
+                );
+            }
+            if (JWT_PROPERTY_KEY.equals(key)) {
+                Map<String, Object> props = Maps.getOrDefault(properties, JWT_PROPERTY_KEY, Map.of());
+                // Check that all required properties are specified.
+                for (String property: JWT_PROPERTIES) {
+                    if (props.containsKey(property) == false) {
+                        throw new IllegalArgumentException(
+                            String.format(Locale.ENGLISH, "JWT property must have property '%s'", property)
+                        );
+                    }
+                }
+                // Check that no unknown properties are specified.
+                for (String jwtProperty: props.keySet()) {
+                    if (JWT_PROPERTIES.contains(jwtProperty) == false) {
+                        throw new IllegalArgumentException(
+                            String.format(Locale.ENGLISH, "'%s' is not a valid JWT property", jwtProperty)
+                        );
+                    }
+                }
+            }
```

Comment:
Looks like, the `password` mapping logic is gone here.

Reply:
https://github.com/crate/crate/pull/15515#discussion_r1479519811 or it's something different?